Rakefile

```diff
@@ -2,6 +2,11 @@ require 'rubocop/rake_task'
 require 'rake/testtask'
 require 'rspec/core/rake_task'
 
+def valid_gem?(input_gem)
+  valid_gems = %w(core grpc nxapi)
```

Comment:
Since `cisco_os_shim/nxapi` and `cisco_os_shim/grpc` gems depend on `cisco_os_shim/core`, we build the `cisco_os_shim/core` gem no matter what value is in `ENV['GEM']`.

After that, the `next` statement will leave the task early if we only wanted to build the `cisco_os_shim/core` gem.

Otherwise, we build either the `grpc` or `nxapi` gem depending on the value of `gem_target`.

This PR won't change node_utils behavior.  I'm adding this feature so that Travis can build the `cisco_os_shim/core` gem without needing `nxapi` or `grpc` dependencies.

Example run for nxapi:

```
~/cisco-nxapi$ rake build GEM=nxapi
  Successfully built RubyGem
  Name: cisco_os_shim
  Version: 1.0.0.pre.dev
  File: cisco_os_shim-1.0.0.pre.dev.gem
WARNING:  prerelease dependency on cisco_os_shim (= 1.0.0.pre.dev) is not recommended
WARNING:  See http://guides.rubygems.org/specification-reference/ for help
  Successfully built RubyGem
  Name: cisco_os_shim-nxapi
  Version: 1.1.0.pre.dev
  File: cisco_os_shim-nxapi-1.1.0.pre.dev.gem
```